Druid/DRBasicBlock.class.st

```diff
@@ -345,8 +345,7 @@ DRBasicBlock >> greater: operand1 than: operand2 [
 { #category : #testing }
 DRBasicBlock >> hasFinalInstruction [
 
-	^ (endInstruction notNil and: [ endInstruction isNoop not ]) or: [ 
-		  instructions anySatisfy: [ :i | i isFrameReturn ] ]
```

Comment:
This doesn't break the inlining!? Woow

Reply:
No idea. I broke a couple of tests, do not seem closely related to inlining


DRBytecodeGeneratorTest
 ✗ #testMethodRecompilationWithProfile (82ms)

DRLinearizationTest
 ✗ #testBuildCFGHasAllBlocks (69ms)
 ✗ #testBuildCFGHasAllBlocksWhenEarlyMethodReturns (54ms)
 ✗ #testBuildCFGHasAllBlocksWithTwoNonLocalReturns (59ms)
 ✗ #testBuildCFGHasAllBlocks (63ms)
 ✗ #testBuildCFGHasAllBlocksWhenEarlyMethodReturns (53ms)
 ✗ #testBuildCFGHasAllBlocksWithTwoNonLocalReturns (53ms)

DRCogitCanonicaliserTest
 ✗ #testSimplifyAdditionOfAddition (2ms)
 ✗ #testSimplifyLoadAdditionOfAddition (1ms)

Reply:
So, testMethodRecompilationWithProfile is probably related, I tried to fixed it but it is too tangled on how the bytecode is generated...

Reply:
Ahhh that one can be skipped, it was the last experiment with DrOpal so everything is very fragile. It's ok!